### PR TITLE
Add `updated_by` to API Keys

### PR DESCRIPTION
**Pull request by johnswanson**
This required a bit of modification of the DB.

First, I had named the column meant to hold the ID of the user who last updated an API key as `updated_by`. But it seemed like `updated_by` was the natural field to hold the user name that FE wanted, and I didn't want to clobber an existing field with `hydrate`.

So, I renamed the existing column to `updated_by_id` and used `updated_by` to hold the hydrated user that updated the ApiKey.

Second, for consistency, I renamed `created_by` to `creator_id`.

The logic for presenting an ApiKey was getting a bit complicated and spread out over many different endpoints, so I pulled it out into its own function that's called from almost every endpoint defined in the `metabase.api.api-key` namespace. Note that it calls `(t2/hydrate api-key :updated_by :group_name)` - but I verified that this will not result in additional DB calls if the model was already hydrated.

### CODE REVIEW

**Comment by johnswanson** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/37014?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `api-key-crud`
  * **PR #37014** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37014?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #37051** <a href="https://app.graphite.dev/github/pr/metabase/metabase/37051?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/37014?utm_source=stack-comment).

File: resources/migrations/001_update_migrations.yaml
Comment (noahmoss) [inline resources/migrations/001_update_migrations.yaml:4430]: We actually have `creator_id` as the column name in some tables, like `report_dashboard`. Maybe use that instead for consistency? 

You could also call the other one `updater_id` but it sounds a little less natural

File: src/metabase/models/api_key.clj
Comment (noahmoss) [inline src/metabase/models/api_key.clj:83]: Wouldn't this _not_ modify `:updated_by_id` once it's set for the first time? The anonymous fn takes the old value as an arg, so it'll always be used preferentially, right?

Comment (noahmoss) [reply src/metabase/models/api_key.clj:83]: Also — I think generally we're trying to move away from using the dynamic vars in `metabase.api.common` in non-API namespaces to make things more explicit and REPL-friendly. (Granted, we have a lot of it at the moment, but there's been some recent discussions about curbing it, because it's annoying to constantly have to bind `*current-user-id*` during development)

I won't block the PR over this, but I think it would be slightly better to keep these in the API namespace like they were before. Or add a functions in the model namespace to create/update API keys which take a user ID as an argument, similar to `metabase.models.card/create-card!`

**Review by noahmoss — APPROVED** [review]
Stamping to unblock, but have a couple comments that I'm confident you can address

File: resources/migrations/001_update_migrations.yaml
Comment (johnswanson) [reply resources/migrations/001_update_migrations.yaml:4430]: I thought about this before and was torn between:
- `created_by_id` is consistent with `updated_by_id` in the same table, but inconsistent with other tables
- `creator_id` is consistent with other tables, but then not consistent with `updated_by_id`
But I think you're right and I'll change this - the inconsistency between `creator_id` and `updated_by_id` is pretty minimal. And if we wanted to hydrate the `creator` of the ApiKey, we could do so already (since it's already set up), while we'd need another line of code to tell t2 how to automagically hydrate the `created_by`.

File: src/metabase/models/api_key.clj
Comment (johnswanson) [reply src/metabase/models/api_key.clj:83]: Good catch - and keeping `api` out of the models does seem more hygienic, I'll do that.

**Pull request merged**
